python-runner/runner.py

```diff
@@ -78,11 +79,133 @@ class PipelineRequest(BaseModel):
     config_map.close()
 
 
-def apply_pipeline(record: dict, pipeline: dict, preview_step=None):
+def apply_pipeline(record: dict, pipeline: dict):
```

Comment:
Are these changes intended to be part of this PR?

Reply:
Yes. This PR introduced the new method `preview_pipeline`, which is only used for the `/preview` endpoint. It performs a couple of tasks that are not needed for the deployment and might have a slightly negative impact on the performance.